### DISCUSSION TITLE
Compilation and deployment time

### DISCUSSION THREAD

**Discussion by project-x51** [dotnet/maui#24299] [Q&A]
Hi,

I'm trying to figure out how to speed up the compilation and deployment time working on my .NET Maui project. 

I have temporarily moved from my normal workstation (i9 13900K, 96GB 6600 MHz RAM, 2 x PCIe 4 M.2 SSDs in RAID 0) to my backup workstation (i9 13700, 96GB 6600 MHz RAM, 1 x PCIe M2.SSD). I have noticed a marked increase in deployment time which I can only put down to the increase in speed due to the M.2 SSD RAID on my primary workstation.

When I look at the loads on task manager, I hardly see any load on my disk, and only a short amount of time where the CPU hits 100%. What is happening most of the time? is it disk constrained (I think so) and the task manager isn't showing it. Or is something else going on? Is it bandwidth constrained, or IOPS (I think its IOPS)?

I'm thinking of expanding the RAID 0 to 4 drives as my understanding is this should increase the IOPS by almost double over 2. I expect the bandwidth will also increase but by a smaller amount. Before I do, I'd like to hear what people think is the bottleneck. Also interested in peoples experience building a 4 drive RAID 0 from m.2 SSDs using the Intel Rapid Storage controller. And yes, I know RAID 0 offers no redundancy. Just give me speed, all of it!

Robin

**Comment by MartyIX** [comment]
I think that creating a binlog and analysing it is the standard way to answer your question. See https://github.com/dotnet/msbuild/blob/main/documentation/wiki/Providing-Binary-Logs.md

**Reply by project-x51** [reply]
Interesting, this is clearly going to tell me what the build was doing and how long it took doing each task. But will it tell me which compute resources were needed to do each task?

**Reply by MartyIX** [reply]
The [viewer](https://github.com/KirillOsenkov/MSBuildStructuredLog?tab=readme-ov-file#install) can give you a timeline or a tracing view how long build operations took which is an approximation of resources needed. So it's more basic. I'm not sure about details, I'm not an author nor an expert on this topic. I just know people use it quite often to debug long msbuild tasks and to understand ordering of msbuild tasks.
